synapse/replication/http/_base.py

```diff
@@ -106,6 +106,25 @@ def __init__(self, hs):
 
         assert self.METHOD in ("PUT", "POST", "GET")
 
+        self._replication_secret = None
+        if hs.config.worker.worker_replication_secret:
+            self._replication_secret = hs.config.worker.worker_replication_secret
+
+    def check_auth(self, request) -> None:
+        # Get the authorization header.
+        auth_headers = request.requestHeaders.getRawHeaders(b"Authorization")
+
+        if len(auth_headers) > 1:
+            raise RuntimeError("Too many Authorization headers.")
```

Comment:
In this case the client will receive a 500 error. I'm not sure if there's a "better" exception to raise that would get passed back. But since this is a misconfiguration it isn't necessarily that the client was unauthorized or such.

Reply:
Yeah, and it will send exceptions to sentry like this :+1: